docs/providers/aws/events/apigateway.md

```diff
@@ -185,6 +185,43 @@ functions:
             identityValidationExpression: someRegex
 ```
 
+You can also configure an existing Cognito User Pool as the authorizer, as shown
+in the following example:
+
+```yml
+# serverless.yml
+functions:
+  create:
+    handler: posts.create
+    events:
+      - http:
+          path: posts/create
+          method: post
+          authorizer:
+            arn: arn:aws:cognito-idp:us-east-1:xxx:userpool/us-east-1_ZZZ
+```
+
+By default the `sub` claim will be exposed in `events.cognitoPoolClaims`, you can add extra claims like so:
+
+```yml
+# serverless.yml
+functions:
+  create:
+    handler: posts.create
+    events:
+      - http:
+          path: posts/create
+          method: post
+          integration: lambda
+          authorizer:
+            arn: arn:aws:cognito-idp:us-east-1:xxx:userpool/us-east-1_ZZZ
+            claims:
+              - email
+              - nickname
+```
+
+You must use `integration: lambda` to have any claims exposed.
```

Comment:
if it's the former, could you please throw a Serverless error that covers this? so that users don't have to wait during deployment to find out about the limitation?